website/static/js/filters.js

```diff
@@ -0,0 +1,149 @@
+                            applied
+                groups:     -a dictionary of filter button groups
+                            -key: the ID for the group of filter buttons
+                            -value: a dictionary of options with the following format:
+                                - {
+                                    filter: (selector to be used for filtering),
+                                    type: (currently the options are 'text' or 'checkbox'),
+                                    buttons: {
+                                        (buttonID): (value to match) (if type is 'text' the value is the string that
+                                        should be matched.  if type is 'text' the value is the boolean that should be
+                                        matched),
+                                        ....
+                                    }
+                                }
+                inputs:     -a dictionary of the search fields
+                            -key: the ID for the search field
+                            -value: the selector that this field should search against
+         */
+        var settings = $.extend({
+            items: ['.items']
+        }, options);
+
+        var itemsSelector = settings.items.join();
+
+        var filterConstraints = {};
+        var searchConstraints = {};
+
+        var search = function(el) {
+            for (var key in searchConstraints) {
+                if (searchConstraints.hasOwnProperty(key)) {
+                    var content = jQuery(el.querySelector(key)).text().toLowerCase();
+                    var exists = content.indexOf(searchConstraints[key]);
+                    if (exists === -1) {
+                        return false;
+                    }
+                }
+            }
+            return true;
+        };
+
+        var filter = function(el) {
+            for (var key in filterConstraints) {
+                if (filterConstraints.hasOwnProperty(key)) {
+                    var selector = settings.groups[key].filter;
+                    var type = settings.groups[key].type;
+                    var match = filterConstraints[key];
+                    if (type === 'text') {
+                        var content = jQuery(el.querySelector(selector)).text();
+                        if (match.indexOf(content) === -1) {
+                            return false;
+                        }
+                    }
+                    else if (type === 'checkbox') {
+                        if (match.indexOf(el.querySelector(selector).checked) === -1) {
+                            return false;
+                        }
+                    }
+                }
+            }
+            return true;
+        };
+
+        var toggle = function() {
+            var activeItems = $(itemsSelector);
+            $(itemsSelector).each(function() {
+                var self = this;
+                if (!search(self)) {
```

Comment:
Since both conditions do the same thing, can you change it to `if (!search(self) || !filter(self)`?